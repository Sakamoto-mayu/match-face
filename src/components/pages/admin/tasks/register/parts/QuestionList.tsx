'use client'
import questionData from "@/const/answerEdit"
import CheckBox from "@/components/ui/checkbox/CheckBox"

const QuestionList = () => {
  const tableDefaultClassName = "border-2 border-deep-gray px-4 py-2 text-center"
  return (
    <div className="w-full mx-auto">
      <table className="border-collapse border-2 border-deep-gray text-sm mx-auto w-full">
        <thead className="bg-light-gray">
          <tr>
            <th className={`${tableDefaultClassName} w-1/12`}></th>
            <th className={`${tableDefaultClassName} w-2/12`}>最終編集日</th>
            <th className={`${tableDefaultClassName} w-3/12`}>案件名</th>
            <th className={`${tableDefaultClassName}`}>質問内容</th>
          </tr>
        </thead>
          {questionData.map((projects) => (
            <tbody key={`pjId_${projects.id}`}>
              {projects.questions.map((questions) => (
                <tr key={`userId_${questions.question_id}`}>
                  <td className={`${tableDefaultClassName}`}>
                    <CheckBox />
                  </td>
                  <td className={`${tableDefaultClassName}`}>{projects.edit_date}</td>
                  <td className={`${tableDefaultClassName}`}>{projects.project_name}</td>
                  <td className={`${tableDefaultClassName}`}>{questions.question}</td>
                </tr>
              ))}
            </tbody>
          ))}
      </table>
    </div>
  )
}

export default QuestionList
