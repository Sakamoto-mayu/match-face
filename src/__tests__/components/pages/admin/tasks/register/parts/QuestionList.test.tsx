import QuestionList from "@/components/pages/admin/tasks/register/parts/QuestionList";
import { render } from "@testing-library/react";

describe("Question.tsx", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("QuestionListのスナップショット", () => {
    const { container } = render(<QuestionList />);
    expect(container).toMatchSnapshot();
  });
})
