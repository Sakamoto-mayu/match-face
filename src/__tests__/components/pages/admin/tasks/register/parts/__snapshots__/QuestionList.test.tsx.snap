// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question.tsx QuestionListのスナップショット 1`] = `
<div>
  <div
    class="w-full mx-auto"
  >
    <table
      class="border-collapse border-2 border-deep-gray text-sm mx-auto w-full"
    >
      <thead
        class="bg-light-gray"
      >
        <tr>
          <th
            class="border-2 border-deep-gray px-4 py-2 text-center w-1/12"
          />
          <th
            class="border-2 border-deep-gray px-4 py-2 text-center w-2/12"
          >
            最終編集日
          </th>
          <th
            class="border-2 border-deep-gray px-4 py-2 text-center w-3/12"
          >
            案件名
          </th>
          <th
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            質問内容
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            <input
              checked=""
              class="w-4 h-4 accent-orange cursor-pointer"
              type="checkbox"
              value="問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１"
            />
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            2023-7-6
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            test1
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１
          </td>
        </tr>
        <tr>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            <input
              class="w-4 h-4 accent-orange cursor-pointer"
              type="checkbox"
              value="問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２"
            />
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            2023-7-6
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            test1
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２
          </td>
        </tr>
        <tr>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            <input
              class="w-4 h-4 accent-orange cursor-pointer"
              type="checkbox"
              value="問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３"
            />
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            2023-7-6
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            test1
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
